### PR TITLE
feat(FX-4106): use `artworksConnection` for auctions rail on home page

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves [FX-4106]

### Description
Currently we are using `saleArtworksConnection` for "Auctions" rail on the home page. We want to replace this with `artworksConnection` to simplify development and support in the future (as we did it for [FX-4089](https://artsyproduct.atlassian.net/browse/FX-4089)).

rail-1

### QA Test Case(s)

| Test Case | Feature | Environment | Acceptance Criteria | Setup Instructions/Link |
| --------- | :-----: | ----------: | ------------------: | ----------------------: |
|           |         |             |                     |                         |

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- use `artworksConnection` for "Auctions" rail on home page - dimatretyak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[FX-4106]: https://artsyproduct.atlassian.net/browse/FX-4106?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by gkartalis — APPROVED** [review]
noice 🚝

File: src/app/Scenes/Home/Components/SalesRail.tsx [hunk 1]
```diff
@@ -179,6 +189,15 @@ export const SalesRailFragmentContainer = createFragmentContainer(SalesRail, {
             }
           }
         }
+        artworksConnection(first: 3) {
+          edges {
+            node {
+              image {
+                url(version: "large")
```
Comment (dblandin) [inline src/app/Scenes/Home/Components/SalesRail.tsx:196]: **question / non-blocking**: Probably not relevant to this PR but does this image quality work well in 2x/3x contexts? Are we able to use srcSet on a resized image instead?

**Review by dblandin — APPROVED** [review]
LGTM! One non-blocking question on image quality and srcSet.

File: src/app/Scenes/Home/Components/SalesRail.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/app/Scenes/Home/Components/SalesRail.tsx:196]: as far as I understand, `srcSet` is not used anywhere in Eigen (there is only one place where [this field is requested](https://github.com/artsy/eigen/blob/d2a56f307424c84928e73e029b8a92a88ae50831/src/app/Components/ArtworkRail/ArtworkRailCard.tsx#L157), but in the end it is not used)

Comment (dimatretyak) [reply src/app/Scenes/Home/Components/SalesRail.tsx:196]: also as I understand in `resized.src` we will also get a `large` version of the image

Artsy - Metaphysics (Staging) – Sandbox 2022-07-26 19-47-37

Comment (dblandin) [reply src/app/Scenes/Home/Components/SalesRail.tsx:196]: You can pass a version to `resized`. Not sure why it defaults to "large" 🤷🏼 

```graphql
resized(version: "normalized")
```

Comment (dimatretyak) [reply src/app/Scenes/Home/Components/SalesRail.tsx:196]: If you don't mind, I will create a ticket in the backlog and then at one of the refinement sessions we can discuss this moment with the team. What do you think about this?

Comment (dblandin) [reply src/app/Scenes/Home/Components/SalesRail.tsx:196]: > as far as I understand, `srcSet` is not used anywhere in Eigen

Yes, that does seem to be the case! I wonder why as it seems like the ReactNative image component has some support multiple images with different scales: https://reactnative.dev/docs/image#source

Definitely not blocking but something I'm interested in learning more about! We use `srcSet` and image URIs denoting different scales (currently 1x and 2x) often in Force.

Comment (dblandin) [reply src/app/Scenes/Home/Components/SalesRail.tsx:196]: > If you don't mind, I will create a ticket in the backlog

@dimatretyak Sounds good! I think Mobile Platform is also having similar discussions around images.

Comment (dimatretyak) [reply src/app/Scenes/Home/Components/SalesRail.tsx:196]: [Jira ticket](https://artsyproduct.atlassian.net/browse/FX-4110). Feel free to change/add it

**Pull request merged**